x-pack/plugins/index_management/public/application/components/component_templates/component_template_list/component_template_list.tsx

```diff
@@ -141,13 +145,13 @@ export const ComponentTemplateList: React.FunctionComponent<Props> = ({
   let content: React.ReactNode;
 
   if (isLoading) {
-    content = (
-      <SectionLoading data-test-subj="sectionLoading">
+    return (
```

Comment:
I've never been a big fan of this `content` var declared on top and assigned down.

Can we write this slightly differently, with simply an early return?

```js
if (isLoading) {
  return (
    <PageLoading data-test-subj="sectionLoading">
      <FormattedMessage
        id="xpack.idxMgmt.home.componentTemplates.list.loadingMessage"
        defaultMessage="Loading component templates…"
      />
    </PageLoading>
  );
}

let content: React.ReactNode;

if (data?.length) {

...
```

Reply:
We could but it would require duplicating the wrapping `<div className={APP_WRAPPER_CLASS} data-test-subj="componentTemplateList">` element in several places. I think we can find a more elegant solution later when we rearchitect our apps to adhere to a consistent pattern, so I'm going to defer this for now.

Reply:
It all goes to down to preferences but I never have to use this pattern of declaring an empty `content` and assign it in multiple place to finally render it down. I much prefer early return pattern

```js
const SomeComponent = () => {
  if (this) {
    return <MyComp />;
  }
  
  if (that) {
    return <MyOtherComp />;
  }
  
  const renderMainContent = () => {
    if (someFlag) {
      return <div>Early return</div>;
    }
  
    return <div>Finally</div>;
  }
  
  // otherwise
  return <div className="someClassWeWantEverywhere">{renderMainContent()}</div>
}
```

Reply:
Ah, I see what you're saying. Yes I prefer that, too. We'll apply this in the future when we rearchitect our apps to adhere to a consistent pattern.